### PR TITLE
GTFS aggregator scraper v2

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

[Design doc](https://docs.google.com/document/d/1uVgtu7wazRP7G4hT-AgCFpPIjH0MgZo6GLKxKCP_eHM/edit)

Creates scraping code, image, Airflow DAG, external tables, and initial dbt models. We may want to use the legacy data to provide past coverage, but only URLs present in agencies.yml were checked.

Resolves or makes the following PRs redundant
https://github.com/cal-itp/data-infra/issues/1825
https://github.com/cal-itp/data-infra/issues/1826
https://github.com/cal-itp/data-infra/issues/2078

Makes progress towards these
https://github.com/cal-itp/data-infra/issues/1999
https://github.com/cal-itp/data-infra/issues/2015

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?
Locally, up through dbt tests.

### CODE REVIEW

**Comment by owades** [comment]
I want to confirm the table will work for building a check - @atvaccaro can you take a logic at how I'm hoping to use join a feed_guideline_index table to `stg_gtfs_quality__scraped_urls`? 

```
      FROM feed_guideline_index t1
      LEFT JOIN dim_feed_info t2
        ON t1.feed_key = t2.feed_key
      LEFT JOIN dim_gtfs_datasets t3
        ON t2.base64_url = t3.base64_url
      LEFT JOIN daily_scraped_urls AS t4
        ON t3.uri = t4.feed_url_str
```

**Comment by atvaccaro** [comment]
> I want to confirm the table will work for building a check - @atvaccaro can you take a logic at how I'm hoping to use join a feed_guideline_index table to `stg_gtfs_quality__scraped_urls`?
> 
> ```
>       FROM feed_guideline_index t1
>       LEFT JOIN dim_feed_info t2
>         ON t1.feed_key = t2.feed_key
>       LEFT JOIN dim_gtfs_datasets t3
>         ON t2.base64_url = t3.base64_url
>       LEFT JOIN daily_scraped_urls AS t4
>         ON t3.uri = t4.feed_url_str
> ```

Ah this is going to change quite soon because of versioned Airtable, but yes that looks good. (the only change will be adding a timestamp to the `dim_gtfs_datasets` join)

**Review by owades — APPROVED** [review]

File: airflow/dags/scrape_feed_aggregators/METADATA.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-description: "Check common GTFS feed aggregators for canonical URLs in agencies.yml"
+description: "Scrape GTFS (and other) feed URLs present in validators"
```
Comment (SorenSpicknall) [inline airflow/dags/scrape_feed_aggregators/METADATA.yml:1]: Besides this, do we have any replacement for the deleted README that was in `jobs/gtfs-aggregator-checker`? Should we replace that README 1:1 with a new version in `jobs/gtfs-aggregator-scraper`

File: jobs/gtfs-aggregator-scraper/gtfs_aggregator_scraper.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+class GTFSFeedAggregator(str, Enum):
+    mobility_database = "mobility_database"
+    transitland = "transitland"
+
+
+class GTFSFeedAggregatorPresence(BaseModel):
+    key: str
+    name: Optional[str]
+    feed_url_str: str
+    # feed_url: HttpUrl
```
Comment (SorenSpicknall) [inline jobs/gtfs-aggregator-scraper/gtfs_aggregator_scraper.py:54]: Should this comment stay in?

File: airflow/dags/scrape_feed_aggregators/METADATA.yml [hunk 3]
```diff
@@ -9,7 +9,6 @@ default_args:
     email:
       - "evan.siroky@dot.ca.gov"
       - "andrew.v@jarv.us"
-      - "hunter.owens@dot.ca.gov"
       - "jameelah.y@jarv.us"
       - "laurie.m@jarv.us"
```
Comment (SorenSpicknall) [inline airflow/dags/scrape_feed_aggregators/METADATA.yml:13]: Should soren.s@jarv.us be added to this list, as well?

**Review by SorenSpicknall — COMMENTED** [review]
Just a few small questions on first pass.

File: airflow/dags/scrape_feed_aggregators/METADATA.yml (hunk 1 above)
Comment (atvaccaro) [reply airflow/dags/scrape_feed_aggregators/METADATA.yml:1]: Eh not really, I don't think a readme is that useful for this

File: jobs/gtfs-aggregator-scraper/gtfs_aggregator_scraper.py (hunk 2 above)
Comment (atvaccaro) [reply jobs/gtfs-aggregator-scraper/gtfs_aggregator_scraper.py:54]: ah yeah let me remove

**Review by SorenSpicknall — APPROVED** [review]

**Pull request merged**